### DISCUSSION TITLE
Miden Assembly Playground

### DISCUSSION THREAD

**Discussion by timgestson** [0xMiden/miden-vm#175] [Show and tell]
Since Miden and Winterfell can be compiled with `no_std`, they can run via wasm in the browser. When I chatted with Bobbin, we briefly talked about the idea of a Miden Playground. I played around with the idea a bit to get more familiar with the assembly language and vm semantics. Here is the proof of concept:

[Playground](https://timgestson.github.io/miden-assembly-playground/)

[Source Code](https://github.com/timgestson/miden-assembly-playground)

In the future we could add Inputs, Metrics about Execution and Proofs, and Example programs.

**Comment by grjte** [comment]
This is really cool!

> In the future we could add Inputs, Metrics about Execution and Proofs, and Example programs

In particular, it would be great to see the cycle count. Tying this together with #138 would be useful

**Comment by bobbinth** [comment]
Very cool! Thank you for putting it together! My dream is that at some point we'll have something like [evm.codes](https://www.evm.codes/playground) for Miden VM, and this is a great start!

**Comment by itzmeanjan** [comment]
This is good :+1:
